### PR TITLE
Feature/16 registerpage info pages (with slider)

### PR DESCRIPTION
**Pull request by Yarin18**

### CODE REVIEW

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt [hunk 1]
```diff
@@ -263,16 +283,217 @@ fun PermissionStep(onAction: (RegisterScreenUiAction) -> Unit) {
...
 
+        SkipTextButton { onAction(RegisterScreenUiAction.OnNextClickedAction) }
+
+
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun InfoStep(onAction: (RegisterScreenUiAction) -> Unit) {
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt]: This is a giant function, maybe extract the others into different files

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt [hunk 2]
```diff
@@ -362,3 +611,17 @@ fun NextButton(onAction: (RegisterScreenUiAction) -> Unit) {
     )
 }
 
+@Composable
+fun SkipTextButton(
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt]: Extract this to a different file

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt [hunk 3]
```diff
@@ -349,6 +570,34 @@ fun CircularProfilePicture(
...
+    }
+
+    Row {
+        LevelColumn(R.string.level_3, R.string._2_dagen_op_voorhand_reserveren, 135.dp)
+        LevelColumn(R.string.level_4, R.string._3_dagen_op_voorhand_reserveren, 135.dp)
+    }
+}
+
+@Composable
+fun LevelColumn(level: Int, explanation: Int, boxHeight: Dp) {
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt]: Extract this to a different file

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt [hunk 4]
```diff
@@ -349,6 +570,34 @@ fun CircularProfilePicture(
     }
 }
 
+@Composable
+fun LevelSquares() {
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt]: Extract this to a different file

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt [hunk 5]
```diff
@@ -263,16 +283,217 @@ fun PermissionStep(onAction: (RegisterScreenUiAction) -> Unit) {
...
+                textAlign = TextAlign.Center
+            ) { onAction(action) }
+        }
+    }
+}
+
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun BottomPagerIndicator(
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt]: Extract this to a different file

File: composeApp/src/main/res/drawable/laadbeurt_afstaan.png
Comment (DanneD-Wisemen) [inline composeApp/src/main/res/drawable/laadbeurt_afstaan.png:1]: Keep the naming in english

File: composeApp/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -24,5 +24,24 @@
     <string name="meldingen_toestaan">Meldingen toestaan</string>
     <string name="meldingen_info">Charge Hub helpt je eraan denken wanneer het jouw beurt is om te laden</string>
     <string name="overslaan">Overslaan</string>
+    <string name="laadbeurten">Laadbeurten</string>
+    <string name="reservatie_info">Reserveer een laadbeurt door te kiezen uit snelle reservaties op de thuispagina of eentje aan te maken!</string>
+    <string name="laadbeurt_afstaan">Laadbeurt afstaan</string>
+    <string name="laadbeurt_afstaan_info">Elke laadbeurt kan afgestaan worden tot 15 min. voor hij van start gaat. Laattijdig afstaan zorgt voor een daling in level.</string>
+    <string name="level_systeem">Level systeem</string>
```
Comment (DanneD-Wisemen) [inline composeApp/src/main/res/values/strings.xml]: Keep the string keys english

File: composeApp/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -24,5 +24,24 @@
...
+    <string name="level_system_title">Je level wordt beïnvloedt door</string>
+    <string name="stiptheid">Stiptheid</string>
+    <string name="stiptheid_details">Op tijd laadbeurten beginnen en je voertuig verplaatsen.</string>
+    <string name="hoffelijkheid">Hoffelijkheid</string>
+    <string name="hoffelijkheid_details"><![CDATA[Ruilverzoeken accepteren & laadbeurten vroegtijdig of laattijdig afstaan.]]></string>
+    <string name="level_systeem_2_uitleg">Hoe hoger je huidige level, hoe meer functies je vrijspeelt zoals</string>
+    <string name="prioriteit">Prioriteit</string>
+    <string name="level_1">Level 1</string>
```
Comment (DanneD-Wisemen) [inline composeApp/src/main/res/values/strings.xml]: Try and do this with a paramater so you don't have to create a string resource for every level

File: composeApp/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -24,5 +24,24 @@
...
+    <string name="level_2">Level 2</string>
+    <string name="level_3">Level 3</string>
+    <string name="level_4">Level 4</string>
+    <string name="ruilen">Ruilen</string>
+    <string name="_3_dagen_op_voorhand_reserveren">3 dagen op voorhand reserveren</string>
+    <string name="_2_dagen_op_voorhand_reserveren">2 dagen op voorhand reserveren</string>
```
Comment (DanneD-Wisemen) [inline composeApp/src/main/res/values/strings.xml]: Same as the level comments

**Review by DanneD-Wisemen — CHANGES_REQUESTED** [review]

File: composeApp/build.gradle.kts [hunk 9]
```diff
@@ -36,6 +36,7 @@ kotlin {
             implementation(libs.androidx.navigation.compose)
             implementation(libs.raamcosta.core)
             implementation(libs.androidx.material3.extended)
+            implementation (libs.accompanist.pager)
```
Comment (DanneD-Wisemen) [inline composeApp/build.gradle.kts]: Space could be removed

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt [hunk 10]
```diff
@@ -70,6 +95,7 @@ fun RegisterScreenPreview() {
     }
 }
 
+@RequiresApi(Build.VERSION_CODES.TIRAMISU)
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt:84]: Why is this needed? This way your screen can only run on Android 13+ I think

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/register/InfoStepComponents.kt [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+            style = TextStyles.infoSubTitle,
+            textAlign = TextAlign.Center,
+            modifier = Modifier.padding(horizontal = 16.dp)
+        )
+        Spacer(modifier = Modifier.weight(1f))
+    }
+}
+
+@Composable
+fun InfoPageOne() {
```
Comment (DanneD-Wisemen) [inline composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/register/InfoStepComponents.kt]: Maybe rename this to what info is shown

**Review by DanneD-Wisemen — CHANGES_REQUESTED** [review]

File: composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt (hunk 10 above)
Comment (Yarin18) [reply composeApp/src/androidMain/kotlin/com/wisemen/chargehub/ui/screens/RegisterScreen.kt:84]: Not adding the gives me this warning in AndroidStudio

Field requires API level 33 (current min is 24): android.Manifest.permission#POST_NOTIFICATIONS

**Review by DanneD-Wisemen — APPROVED** [review]
